packages/grafana-ui/src/components/VizLegend/types.ts

```diff
@@ -47,5 +47,6 @@ export interface VizLegendItem<T = any> {
   // displayValues?: DisplayValue[];
   getDisplayValues?: () => DisplayValue[];
   fieldIndex?: DataFrameFieldIndex;
+  fieldName?: string;
```

Comment:
we create series that have a different name in the legend than the y field which holds the config (color, unit, hideFrom, etc.)

we cannot rely on the visible legend label to control these series, because we need to match on the y field name, not a user-defined custom series name.